lib/tokenable/revokeable.rb

```diff
@@ -0,0 +1,33 @@
+module Tokenable
+  module Revokable
+    extend ActiveSupport::Concern
+
+    def token_revoked?(revoke_token)
+      !revoke_token.in?(revoke_tokens_hash.keys)
+    end
+
+    def issue_revoke_token
+      issued_token = SecureRandom.hex
+      self.update(
+        revoke_tokens_key => revoke_tokens_hash.merge(
+          issued_token => Time.current
+        )
+      )
+
+      issued_token
+    end
+
+    private
+
+    def revoke_tokens_hash
+      raise Tokenable::Unauthorized.new("#{revoke_tokens_key} field is missing") unless self.has_attribute?(revoke_tokens_key)
+
+      read_attribute(revoke_tokens_key).to_h
+    end
+
+    # TODO: make a config option
+    def revoke_tokens_key
+      :revoke_tokens
```

Comment:
They're not technically revoke tokens, they're salts/secrets that are used to check for validity. The exat opposite of a revoke token in terms of terminology.

We could instead use something like: `tokenable_secret_keys` `tokenable_jwt_verifiers`